node/node.go

```diff
@@ -988,81 +985,6 @@ func (node *Node) NewAddress() (address.Address, error) {
 	return wallet.NewAddress(node.Wallet)
 }
 
-// CreateMiner creates a new miner actor for the given account and returns its address.
-// It will wait for the the actor to appear on-chain and add set the address to mining.minerAddress in the config.
-// TODO: This should live in a MinerAPI or some such. It's here until we have a proper API layer.
-// TODO: add ability to pass in a KeyInfo to store for signing blocks.
-//       See https://github.com/filecoin-project/go-filecoin/issues/1843
-func (node *Node) CreateMiner(ctx context.Context, accountAddr address.Address, gasPrice types.AttoFIL, gasLimit types.GasUnits, pledge uint64, pid libp2ppeer.ID, collateral *types.AttoFIL) (_ *address.Address, err error) {
-
-	// Only create a miner if we don't already have one.
-	if _, err := node.miningAddress(); err != ErrNoMinerAddress {
-		return nil, fmt.Errorf("can only have one miner per node")
-	}
-
-	ctx = log.Start(ctx, "Node.CreateMiner")
-	defer func() {
-		log.FinishWithErr(ctx, err)
-	}()
-	pubKey, err := node.getMinerActorPubKey()
-	if err != nil {
-		return nil, err
-	}
-
-	smsgCid, err := node.PorcelainAPI.MessageSendWithDefaultAddress(
-		ctx,
-		accountAddr,
-		address.StorageMarketAddress,
-		collateral,
-		gasPrice,
-		gasLimit,
-		"createMiner",
-		big.NewInt(int64(pledge)),
-		pubKey,
-		pid,
-	)
-	if err != nil {
-		return nil, err
-	}
-
-	var minerAddr address.Address
-	err = node.PorcelainAPI.MessageWait(ctx, smsgCid, func(blk *types.Block, smsg *types.SignedMessage,
-		receipt *types.MessageReceipt) error {
-		if receipt.ExitCode != uint8(0) {
-			return vmErrors.VMExitCodeToError(receipt.ExitCode, storagemarket.Errors)
-		}
-		minerAddr, err = address.NewFromBytes(receipt.Return[0])
-		return err
-	})
-	if err != nil {
-		return nil, err
-	}
-
-	// TODO: https://github.com/filecoin-project/go-filecoin/issues/1843
-	blockSignerAddr, err := node.miningOwnerAddress(ctx, minerAddr)
-	if err != nil {
-		return &minerAddr, err
-	}
-
-	err = node.saveMinerConfig(minerAddr, blockSignerAddr)
-	if err != nil {
-		return &minerAddr, err
-	}
-
-	err = node.setupMining(ctx)
```

Comment:
It turns out we never actually needed to call this on Miner Create. Mining Start handles it.